jmetal-algorithm/src/main/java/org/uma/jmetal/algorithm/multiobjective/moead/MOEADD.java

```diff
@@ -250,7 +239,8 @@ public void updateArchive(S indiv) {
             lastFront.add((S) population.get(i));
           }
         }
-        if (rankSolution.getOrDefault(indiv, 0) == (numRanks - 1)) {
+                if (((int) indiv.getAttribute(ranking.getAttributeIdentifier())) == (numRanks - 1)) {
+//        if (rankSolution.getOrDefault(indiv, 0) == (numRanks - 1)) {
```

Comment:
Please, don't comment old code. If it is wrong, delete it. If it is not, keep it. If you don't know, test it. The whole point of versioning the code is to retrieve it if necessary through the commit history, so don't let commented code.